src/TableFunctions/TableFunctionDictionary.cpp

```diff
@@ -42,10 +46,33 @@ void TableFunctionDictionary::parseArguments(const ASTPtr & ast_function, Contex
 ColumnsDescription TableFunctionDictionary::getActualTableStructure(ContextPtr context) const
 {
     const ExternalDictionariesLoader & external_loader = context->getExternalDictionariesLoader();
+    std::string resolved_name = external_loader.resolveDictionaryName(dictionary_name, context->getCurrentDatabase());
+    auto load_result = external_loader.load(resolved_name);
+    if (load_result)
+    {
+        /// for regexp tree dictionary, the table structure will be different with dictionary structure. it is:
+        /// - id. identifier of the tree node
+        /// - parent_id.
+        /// - regexp. the regular expression
+        /// - keys. the names of attributions of dictionary structure
+        /// - values. the values of each attribution
+        const auto dictionary = std::static_pointer_cast<const IDictionary>(load_result);
+        if (dictionary->getTypeName() == "RegExpTree")
```

Comment:
I don't do it because if I include `Dictionaries/RegExpTreeDictionary.h` in TableFunctionDictionary.cpp. The compiler will complain it cannot find `hs.h`. I don't know why, so I did this trick